src/Storages/Kafka/StorageKafkaUtils.cpp

```diff
@@ -223,7 +224,7 @@ void registerStorageKafka(StorageFactory & factory)
                 supported_columns.emplace_back(column.name, column.type);
         }
         // Kafka engine allows only ordinary columns without default expression or alias columns.
-        if (args.columns.getAll() != supported_columns)
+        if (args.columns.getAll() != supported_columns && !args.getLocalContext()->getSettingsRef()[Setting::allow_kafka_use_default_column])
```

Comment:
I think instead of doing the check here, we should add the columns to supported columns. If we stop supporting columns for any other reason, this check would allow them to pass too.

Reply:
OK